src/common/local_file_system.cpp

```diff
@@ -989,7 +989,7 @@ vector<string> LocalFileSystem::Glob(const string &path, FileOpener *opener) {
 			if (previous_directories.empty()) {
 				result.push_back(splits[i]);
 			} else {
-				if (recursive_search && is_last_chunk) {
+				if (recursive_search || is_last_chunk) {
```

Comment:
This should likely be done whenever there was a glob before - not only for the last chunk. For example could we add a test case with a glob like `test/*/dir/*.parquet` (with a constant directory in the middle) and a directory structure like this:

```sql
a/dir/x.parquet
b/dir2/z.parquet
c/dir/y.parquet
```

Reply:
On second thought I think this should just be `is_last_chunk`

Reply:
Yes I think it should already work - but it is not just about working but also about minimizing the number of calls to `FileExists` and `DirectoryExists` (which can be expensive, particularly when many are executed). `recursive_search || is_last_chunk` means we are always calling `FileExists` and `DirectoryExists` when recursive_search is true.